common/src/main/java/org/opensearch/ml/common/connector/ConnectorHttpClientConfig.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.connector;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Getter
+@EqualsAndHashCode
+public class ConnectorHttpClientConfig implements ToXContentObject, Writeable {
+
+    public static final String MAX_CONNECTION_FIELD = "max_connection";
+    public static final String CONNECTION_TIMEOUT_FIELD = "connection_timeout";
+    public static final String READ_TIMEOUT_FIELD = "read_timeout";
+
+    public static final Integer MAX_CONNECTION_DEFAULT_VALUE = Integer.valueOf(30);
+    public static final Integer CONNECTION_TIMEOUT_DEFAULT_VALUE = Integer.valueOf(30000);
+    public static final Integer READ_TIMEOUT_DEFAULT_VALUE = Integer.valueOf(30000);
+
+    private Integer maxConnections;
```

Comment:
why not using primitive `int` instead of `Integer`, it is usually recommended unless trying to support `null` value use case?
Also it can prevent unnecessary boxing/unboxing later on.

Reply:
default value of int is 0. The target of this `maxConnections` or the other values is: is it is null, then it will automatically use the default value of httpclient. If we use int, then for null it will default to 0 which is not aligning with the goal.